### ISSUE TITLE
Feature Request: New default map J tabprev and K tabnext_gt

### ISSUE THREAD

**Issue by samrocketman** [tridactyl/tridactyl#1433]
# Feature request

It would be nice if going to tabs left and right navigated were controlled by `J` and `K` in addition to `gT` and `gt`.

# How I bind it today

Currently, I bind it for myself.  Others can do the same with the following commands.

```
:bind --mode=normal J tabprev
:bind --mode=normal K tabnext_gt
```

I found the commands used with `:help gt` and `:help gT`.

I'm sharing this in case you agree with me; it would be a nice user-friendly default.

# Irrelevant side note

Cheers and thanks a lot for creating tridactyl.  I couldn't do without vimperator and left Firefox a bit.  tridactyl brought me back.

Here's a kitten

![image](https://user-images.githubusercontent.com/875669/55531214-5e9daf00-5677-11e9-8f78-e3c5881f6683.png)

**Comment by samrocketman** [comment]
Thanks!

**Comment by hcraT** [comment]
Hi,
I don't know if this should be better as a new issue if so please
let me know and I will create a new issue.

I find prev and next tab bindings unconfortable as they are now. Even if the
current mapping is logical (`J` is on the left and does tabprev and `K` is on the right
and does tabnext) the "vim" concept of forward/backward (or down/up) is
reversed meaning that usually j means forward and k means backwards.
I cannot recall how it was in vimperator but in vim-vixen for example the
commands are bound in the way I describe.

I would suggest switching the keys but I would like to hear your opinion.

Thanks,
Cheers!

**Comment by bovine3dom** [comment]
@hcraT you can rebind easily with `:bind J tabnext_gt` and `:bind K tabprev`. I'm reluctant to change what are now default binds as it would change them for everyone. Some people will have become used to the binds and will be very confused when they suddenly change.

There are ways around this using our config update mechanisms but I think I need more persuasion (i.e. more people to come in and tell me your way is better).

**Comment by NickHu** [comment]
Hi, I'd just like to emphasise and echo @hcraT's concerns. I have to change this every time I install tridactyl. J as next tab also matches qutebrowser and pentadactyl from Firefox way back.

**Comment by bovine3dom** [comment]
I still think more people would complain if they were upset with the current binds. What we could do is have all current users keep the current binds and give new users the swapped binds, via a config updater.

**Comment by gtback** [comment]
I found this issue because, like @NickHu, I use Tree-style Tabs and find it confusing that `J` moves up and `K` moves down. But I understand that switching the behavior for users used to the current behavior is worse, so I'll just manually re-map the keys myself.

**Comment by bovine3dom** [comment]
@gtback you're asking at a bad time because our config updater has just failed to migrate quite a few users. I'm giving this a blocked label because _in theory_ I'd be happy to swap them round if I believed I could maintain the current settings for current users.

**Comment by samrocketman** [comment]
Whether it's  JK or HL makes no difference to me.  I just figured I would  report a friendlier default to unbound keys.  If I recall vimperator had something else bound  to HL which is why I used JK  then.  It makes sense logically to use HL if they are unbound.

Here's from vimtutor reference.

```
             ^
             k              Hint:  The h key is at the left and moves left.
       < h       l >               The l key is at the right and moves right.
             j                     The j key looks like a down arrow.
             v
```

**Comment by bovine3dom** [comment]
We use H and L for back and forward in history by default like Vimperator. I don't like the idea of leaving things unbound because I think most people won't bind them and will just be confused.

> You could then commit to never using H, J, K, L so that they're always guaranteed available for user customization. I think this is a nice hybrid idea. And because it's an idea no code has to change.

I'm a little confused by this. We currently have default binds for all of the keys you mention.

**Comment by NickHu** [comment]
I think a recent tridactyl update reset my bindings for this; for Googlers, the way to correct this behaviour is to run

```
bind J tabnext
bind K tabprev
```

**Comment by bovine3dom** [comment]
@NickHu you can probably retrieve you config as described in https://github.com/tridactyl/tridactyl/issues/2706 - sorry!

**Comment by NickHu** [comment]
No need! The snippet I pasted is my entire config

**Comment by samrocketman** [comment]
> I'm a little confused by this. We currently have default binds for all of the keys you mention.

Nevermind, I was the confused person.  Don't mind my last reply.

**Comment by samrocketman** [comment]
I've used both Vimperator on Firefox and Vimium on Chrome.  To be clear, J/K for tab switching is a Vimium default.  I understand changing defaults for users can be highly disruptive for those who are used to existing workflow.  I didn't realize all Vim directional keys already had bindings when I created this issue originally.

I'm OK with this being closed if it is undesirable to change the default.  I think the discussion in this issue has been good and the users who want the behavior have the solution here.  If you close it then I understand :slightly_smiling_face:

**Comment by bovine3dom** [comment]
As I said above:

>  in theory I'd be happy to swap them round if I believed I could maintain the current settings for current users.

I'm happy to keep this open, but I can't promise it will happen any time soon.

**Comment by samrocketman** [comment]
Honestly, I couldn't tell it was default :slightly_smiling_face: because I set the bindings (as I originally reported in my description).

So yeah, if it is default you can close it or if you want to support other  peoples' desires in this issue I'm fine either way.

tridactyl has worked for me from the start (being someone who really missed vimperator).

**Comment by pataquets** [comment]
As I currently understand, the issue is fixed (as far as J=tabprev, K=tabnext). I guess they're also default as of 2022, as that's how they're in my config and I've never changed them.

As for inverting J/K for next/prev as mentioned before, I think it could use to have its own issue opened if there is enough interest to avoid confusion with the OP. FWIW, I'm -1 on it being default, for the above mentioned reasons, as they're very natural being left=prev/right=next. I think it would be backwards with the common default of horizontal tabs, but a special (understandable) case for vertical tabbers which should use customization.

In summary, I guess the issue could be closed, couldn't be @samrocketman? Let's help keeping the issues tidy :smile:

**Comment by samrocketman** [comment]
I didn’t notice it was implemented.  Maintainers could have closed it on merge.  Either way I can close it now.

**Comment by samrocketman** [comment]
Looks like it was closed and reopened by @bovine3dom .  Is there any outstanding reason this should remain open?

**Comment by bovine3dom** [comment]
I closed it because I added the binds, then I reopened it because some people complained that they were the wrong way round. I think it's actually the most common complaint I see about Tridactyl...

I'm happy to leave it closed, I think swapping the binds (even only for new users) will make about as many people unhappy as happy.

**Comment by pataquets** [comment]
I agree 100%. Swapping keys would make for a good `tridactylrc` configuration example, at the most. :wink:

**Comment by nfd9001** [comment]
Might not even be a poor idea to throw that in as an example in the newtab page, honestly. Just a little note like "coming from qutebrowser, Vimperator, Pentadactyl, etc? You may prefer for `J` to go to the previous tab and `K` to go to the next tab; just run these two commands: (...)." or similar when explaining motions in the tutorial. The newtab page was the first place I looked, as a new installer. If there's a will for it, I can just write that PR.

**Comment by bovine3dom** [comment]
I don't know how much more we can add to the new tab page - it's already too long in my opinion. It does already include `:bind J tabprev` as an example :)

If you think you can improve them, new tab page is here - https://github.com/tridactyl/tridactyl/blob/e4cef503fcd1ce261a246b2540e826f5cf80401a/src/static/newtab.md - and the tutor starts here - https://github.com/tridactyl/tridactyl/blob/e4cef503fcd1ce261a246b2540e826f5cf80401a/src/static/clippy/1-tutor.md
